api/document/js/schema.ts

```diff
@@ -3,22 +3,33 @@ import { Schema } from 'prosemirror-model';
 const schema = new Schema({
   nodes: {
     doc: {
-      content: '(paragraph | unimplemented_node)+',
+      content: 'block+',
     },
-    paragraph: {
-      content: 'inline*',
+    inline: {
+      content: 'text+',
       toDOM: () => ['p', 0],
     },
-    text: {
-      group: 'inline',
+    para: {
```

Comment:
One funky thing I've noticed about ProseMirror's ecosystem is that sometimes optional commands "test" for the existence of certain kinds of schema elements by querying whether a certain name exists on the current schema.  It seems like a sort of "duck typing", but for the existence of schema features.  For example, even though `prosemirror-example-setup` is just an example, it's got a function called [`buildMenuItems`](https://github.com/ProseMirror/prosemirror-example-setup/blob/master/src/menu.js#L163-L216) that tests to see if certain names exist on the schema, and uses that to infer whether it should provide menu items that manipulate those types of content.

Because of this, in my original prototype I had decided to keep the name for paragraphs as `paragraph` (what the [basic schema](https://github.com/ProseMirror/prosemirror-schema-basic/blob/master/src/schema-basic.js#L11-L18) calls paragraphs) rather than `para` (what our own  schema calls paragraphs).  My assumption was that some of ProseMirror's special default functionality for e.g. splitting a paragraph into two if the user pressed <kbd>Enter</kbd> in the middle of one was based on paragraphs being called `paragraph` in the schema.

However, digging into ProseMirror's [`createParagraphNear`](https://github.com/ProseMirror/prosemirror-commands/blob/1e7f0e005b3d9097ede59dd9bc8d24c282e42521/src/commands.js#L247-L262)  command, it seems like it's calling some weird private method on `Node` called [`defaultContentType`](https://github.com/ProseMirror/prosemirror-model/blob/06af1aa1c814176c56f4c82cef7ee39c87fe27dd/src/node.js#L320-L322) to figure out what the type of a paragraph is, so that's weird, and now I'm just confused. 😕

Reply:
Ah, good point, and pressing enter inside the text isn't creating a new paragraph for me. Unfortunately, switching it back to `paragraph` doesn't seem to resolve. I suspect this is related to the rather different `content` configuration; will follow up with more investigation.